### PR TITLE
Add underscore-to-lodash-native transform

### PR DESCRIPTION
**Pull request by zackargyle**
This transform does a few different things.
  1. Convert all underscore imports/requires to lodash imports

        `const _ = require('underscore') => import _ from 'lodash'`

  2. Remove native equivalents

        `_.forEach(array, fn) => array.forEach(fn)`

  3. Remove unused imports after (2)
  4. Use partial imports from lodash to allow tree-shaking

        `import _ from 'lodash' => import {find, pluck} from 'lodash'`
  5. Alternatively there is a `--split-imports=true` option that will prefer multiple import lines.

        `import _ from 'lodash' => import find from 'lodash/find'; import pluck from 'lodash/pluck'`

**Issues:**
1. Does not check for variables with same name in scope
2. Knows nothing of types, so objects using _ methods will break

### CODE REVIEW

File: transforms/__testfixtures__/underscore-to-lodash-native.output.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Test comment
+import {find} from 'lodash';
+[1, 2].forEach(num => num);
+const test = [1, 2, 3];
+const result = find(test, 2);
```
Comment (lencioni) [inline transforms/__testfixtures__/underscore-to-lodash-native.output.js]: Should this be `test.find(item => item === 2)`?

Comment (zackargyle) [inline transforms/__testfixtures__/underscore-to-lodash-native.output.js]: Totes. I did not even realize that was a thing. lol. I'll fix it up.

Comment (lencioni) [inline transforms/__testfixtures__/underscore-to-lodash-native.output.js]: 👍 I bet there's a bunch more we could convert to native pretty easily. How do you feel about adding a test for every underscore method so we can make sure that they are all converted in the best possible way?

e.g.

```js
const test = [1, 2, 3];
const result = _.max(test);
```

could be:

```js
const test = [1, 2, 3];
const result = Math.max(...test);
```

Comment (zackargyle) [inline transforms/__testfixtures__/underscore-to-lodash-native.output.js]: Math.max is interesting, but would only work if no iteree function was provided. I can add some more tests, though they're all handled the same way, so it's a little redundant. But I'm okay with that.

File: transforms/underscore-to-lodash-native.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  lastIndexOf: 'lastIndexOf',
+  first: (j, identifier) => j.memberExpression(identifier, j.literal(0)),
+  last: (j, identifier) => j.memberExpression(identifier,
+    j.binaryExpression('-',
+      j.memberExpression(identifier, j.identifier('length')),
+      j.literal(1)
+    )
+  ),
+};
+
```
Comment (jdalton) [inline transforms/underscore-to-lodash-native.js]: Just a bit of ⚠️ : Methods like `_.each`, `_.map`, `_.filter`, `_.every`, `_.some`, `_.includes` aren't 💯 equiv to built-ins. They're "collections" methods which means they work with array-likes and objects. That and iteratee shorthands are some of the reasons to use them.

Comment (zackargyle) [inline transforms/underscore-to-lodash-native.js]: Unfortunately without a type system there's no way to know if the variable is an object or array. :( I did, however, mention in the PR description and also in the large code description block about Objects breaking in this case. This is definitely not perfect, and it'd be super cool if we added an option if you use flowtype to use the typing data for more effective results.

Comment (jdalton) [inline transforms/underscore-to-lodash-native.js]: > This is definitely not perfect, and it'd be super cool if we added an option if you use flowtype to use the typing data for more effective results.

Or move the less-safe mods to an option. Other "arrays" methods like `_.indexOf`, `_.lastIndexOf`, `_,findIndex` are a safer transition to built-ins.

**Pull request merged**